Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -19,24 +19,42 @@ struct SymbolGraphLoader {
     private(set) var symbolGraphs: [URL: SymbolKit.SymbolGraph] = [:]
     private(set) var unifiedGraphs: [String: SymbolKit.UnifiedSymbolGraph] = [:]
     private(set) var graphLocations: [String: [SymbolKit.GraphCollector.GraphKind]] = [:]
-    private var dataProvider: DocumentationContextDataProvider
+    private var dataLoader: (URL, DocumentationBundle) throws -> Data
     private var bundle: DocumentationBundle
     private var symbolGraphTransformer: ((inout SymbolGraph) -> ())? = nil
     
+    /// Creates a new symbol graph loader
+    /// - Parameters:
+    ///   - bundle: The documentation bundle from which to load symbol graphs.
+    ///   - dataLoader: A closure that the loader uses to read symbol graph data.
+    ///   - symbolGraphTransformer: An optional closure that transforms the symbol graph after the loader decodes it.
+    init(
+        bundle: DocumentationBundle,
+        dataLoader: @escaping (URL, DocumentationBundle) throws -> Data,
```

Comment:
This trick with the closure is a bit confusing. Are you trying to workaround the same problem as we have in `DocumentationContext` with the new and soon-to-be-deprecated initializers? Maybe we should follow the same pattern we have in `DocumentationContext` with the legacy/new enum and protocol? But for the data loader this time?

Reply:
Correct. 

My reason for using a closure here instead of a legacy/new enum like in the context was that I was trying to isolate then legacy/new knowledge to the context. This code can be simplified after 6.2 is released so at the very least I should add a comment with that information.

Reply:
I added a comment saying that this code can be simplified after 6.2 in https://github.com/swiftlang/swift-docc/pull/1057/commits/bf1d51c4d5f3fdc115959259d7e0e91293d8a685